pkg/resource/deploy/step_generator.go

```diff
@@ -555,7 +555,79 @@ func (sg *stepGenerator) GenerateDeletes() []Step {
 			}
 		}
 	}
-	return dels
+
+	if len(targets) > 0 {
+		logging.V(7).Infof("Planner was asked to only delete '%v'", targets)
+
+		resourcesToDelete := make(map[resource.URN]bool)
+
+		for _, target := range targets {
+			current, has := sg.plan.olds[target]
+			if !has {
+				logging.V(7).Infof("Resource to delete (%v) could not be found in the stack.", target)
+				if strings.Contains(string(target), "$") {
+					sg.plan.Diag().Errorf(diag.GetResourceToDeleteCouldNotBeFoundError(), target)
+				} else {
+					sg.plan.Diag().Errorf(diag.GetResourceToDeleteCouldNotBeFoundDidYouForgetError(), target)
+				}
+				return nil, result.Bail()
```

Comment:
Instead of bailing early, should we process the whole list of targets and bail after if any were missing?

Reply:
(note that as-written we'll be doing a fair amount of work here if, say, only the last resource in the targets list is missing--might be worth scanning the list up-front and then doing the `calculateDependentReplacements` bit iff all targets were present)

Reply:
Done.